dispatch/dispatch.go

```diff
@@ -75,73 +70,12 @@ func (d Dispatcher) Start(_ context.Context) error {
 	}
 	d.SqsClient = sqs.New(sess)
 
-	d.Wg.Add(1)
-	go d.sendEvents()
 	return nil
 
 }
 
-func (d Dispatcher) Dispatch(deviceID string, msg *wrp.Message) error {
-	if deviceID == d.Norn.DeviceID {
-		en := NewEventWithID(msg, deviceID)
-		select {
-		case d.DispatchQueue.Load().(chan *eventWithID) <- en:
-			d.Measures.EventQueueDepthGauge.Add(1.0)
-		default:
-			d.queueOverflow()
-			d.Measures.DroppedQueueCount.Add(1.0)
-		}
-	}
-	return nil
-}
-
-func (d Dispatcher) sendEvents() {
-	defer d.Wg.Done()
-	queue := d.DispatchQueue.Load().(chan *eventWithID)
-	select {
-
-	case en, ok := <-queue:
-		if !ok {
-			break
-		}
-		d.Measures.EventQueueDepthGauge.Add(-1.0)
-
-		d.Mutex.RLock()
-		deliverUntil := time.Unix(0, d.Norn.ExpiresAt)
-		dropUntil := d.DropUntil
-		d.Mutex.RUnlock()
-
-		now := time.Now()
-
-		if now.Before(dropUntil) {
-			d.Measures.DroppedCutoffCounter.Add(1.0)
-		}
-		if now.After(deliverUntil) {
-			d.Measures.DroppedExpiredCounter.Add(1.0)
-		}
-
-		d.Workers.Acquire()
-		d.Measures.WorkersCount.Add(1.0)
-		go d.send(en.Event)
-	}
-	for i := 0; i < d.MaxWorkers; i++ {
-		d.Workers.Acquire()
-	}
-
-	return
-
-}
-
-func (d *Dispatcher) empty(droppedCounter metrics.Counter) {
-	droppedMsgs := d.DispatchQueue.Load().(chan *eventWithID)
-	d.DispatchQueue.Store(make(chan *wrp.Message, d.QueueSize))
-	droppedCounter.Add(float64(len(droppedMsgs)))
-	d.Measures.EventQueueDepthGauge.Set(0.0)
-	return
-}
-
 // called to deliver event
-func (d Dispatcher) send(msg *wrp.Message) {
+func (d Dispatcher) Send(msg *wrp.Message) {
```

Comment:
I remember Jack suggesting separate dispatchers - one for SQS and one for HTTP.  So when the manager gets a new norn, it determines which one to make and then the logic for each is separate into different structs while they both implement the same interface.